File: app/src/main/java/com/relic/data/models/ListingItem.java
package com.relic.data.models;

public class ListingItem {

    // the id is the "full name" of an item
    protected String fullName = "";
    private boolean visited;
    private int userUpvoted;
    public boolean saved;
    public String subreddit = "";

    public int userSubmittedPosition;
    public int userCommentsPosition;
    public int userSavedPosition;
    public int userUpvotedPosition;
    public int userDownvotedPosition;
    public int userGildedPosition;
    public int userHiddenPosition;

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public boolean isVisited() {
        return visited;
    }

    public void setVisited(boolean visited) {
        this.visited = visited;
    }

    public int getUserUpvoted() {
        return userUpvoted;
    }

    public void setUserUpvoted(int userUpvoted) {
        this.userUpvoted = userUpvoted;
    }
}

File: app/src/main/java/com/relic/data/models/ListingItem.kt
package com.relic.data.models

open class ListingItem {

    // the id is the "full name" of an item
    open var fullName = ""
    var visited: Boolean = false
    var userUpvoted: Int = 0
    var saved: Boolean = false
    var subreddit : String? = null

    var userSubmittedPosition: Int = 0
    var userCommentsPosition: Int = 0
    var userSavedPosition: Int = 0
    var userUpvotedPosition: Int = 0
    var userDownvotedPosition: Int = 0
    var userGildedPosition: Int = 0
    var userHiddenPosition: Int = 0
}